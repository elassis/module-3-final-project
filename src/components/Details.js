import React, { useEffect, useState } from 'react';
import { useParams, useNavigate } from 'react-router';
import { useDispatch } from 'react-redux';
import { getData } from '../redux/reducer';

const Details = () => {
  const dispatch = useDispatch();
  const { country } = useParams();
  const dateRequest = new Date();
  const URL = `https://api.covid19tracking.narrativa.com/api/${dateRequest.getFullYear()}-${dateRequest.getMonth()}-${dateRequest.getDate()}/country/`;
  const [fetchedData, setFetchedData] = useState([]);
  const navigate = useNavigate();
  useEffect(() => {
    const getCountryData = async (id) => {
      const response = await fetch(`${URL}${id}`);
      const resp = await response.json();
      const date = Object.keys(resp.dates);
      const countryData = Object.values(resp.dates[date].countries);
      setFetchedData(countryData);
    };
    getCountryData(country);
  }, []);

  const handleBackHome = () => {
    dispatch(getData());
    navigate('/');
  };
  return (
    <div>
      <button type="button" className="home-btn" onClick={handleBackHome}>&#10094;</button>
      <div className="upper-section-details">
        <img src="/world-logo.png" alt="logo" className="details-logo" />
        <p className="country-name-details">
          {country}
        </p>
      </div>
      <h3>detailed information</h3>
      <div className="data-detail">
        <p>Total confirmed</p>
        <p>{fetchedData[0] ? fetchedData[0].yesterday_confirmed : null }</p>
      </div>
      <div className="data-detail">
        <p>Total Deaths</p>
        <p>{fetchedData[0] ? fetchedData[0].today_deaths : null }</p>
      </div>
      <div className="data-detail">
        <p>Total Recovered</p>
        <p>{fetchedData[0] ? fetchedData[0].yesterday_recovered : null }</p>
      </div>
    </div>
  );
};

export default Details;
