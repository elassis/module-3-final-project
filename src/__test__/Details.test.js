import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Details from '../components/Details';
import '@testing-library/jest-dom/extend-expect';

describe('Details testing', () => {
  test('renders Details component correctly', () => {
    const tree = render(<Provider store={store}><Router><Details /></Router></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
