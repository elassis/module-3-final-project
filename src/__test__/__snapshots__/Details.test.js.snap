// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details testing renders Details component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <button
          class="home-btn"
          type="button"
        >
          ❮
        </button>
        <div
          class="upper-section-details"
        >
          <img
            alt="logo"
            class="details-logo"
            src="/world-logo.png"
          />
          <p
            class="country-name-details"
          />
        </div>
        <h3>
          detailed information
        </h3>
        <div
          class="data-detail"
        >
          <p>
            Total confirmed
          </p>
          <p />
        </div>
        <div
          class="data-detail"
        >
          <p>
            Total Deaths
          </p>
          <p />
        </div>
        <div
          class="data-detail"
        >
          <p>
            Total Recovered
          </p>
          <p />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <button
        class="home-btn"
        type="button"
      >
        ❮
      </button>
      <div
        class="upper-section-details"
      >
        <img
          alt="logo"
          class="details-logo"
          src="/world-logo.png"
        />
        <p
          class="country-name-details"
        />
      </div>
      <h3>
        detailed information
      </h3>
      <div
        class="data-detail"
      >
        <p>
          Total confirmed
        </p>
        <p />
      </div>
      <div
        class="data-detail"
      >
        <p>
          Total Deaths
        </p>
        <p />
      </div>
      <div
        class="data-detail"
      >
        <p>
          Total Recovered
        </p>
        <p />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
